checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,406 @@
+     */
+    private static final String OLD_NOT_CALLED_METHODS =
+            "org.checkerframework.checker.builder.qual.NotCalledMethods";
+
+    /**
+     * Create a new CalledMethodsAnnotatedTypeFactory.
+     *
+     * @param checker the checker
+     */
+    public CalledMethodsAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(
+                checker,
+                CalledMethods.class,
+                CalledMethodsBottom.class,
+                CalledMethodsPredicate.class);
+        Set<String> disabledFrameworks = new HashSet<>();
+        if (checker.hasOption(CalledMethodsChecker.DISABLE_BUILDER_FRAMEWORK_SUPPORTS)) {
+            disabledFrameworks.addAll(
+                    Arrays.asList(
+                                    checker.getOption(
+                                                    CalledMethodsChecker
+                                                            .DISABLE_BUILDER_FRAMEWORK_SUPPORTS)
+                                            .split(","))
+                            .stream()
+                            .map(String::toUpperCase)
+                            .collect(Collectors.toList()));
+        }
+        builderFrameworkSupports = new ArrayList<>();
+        enableFramework(CalledMethodsChecker.LOMBOK_SUPPORT, disabledFrameworks);
+        enableFramework(CalledMethodsChecker.AUTOVALUE_SUPPORT, disabledFrameworks);
+
+        if (!disabledFrameworks.isEmpty()) {
+            StringJoiner sj = new StringJoiner(", ");
+            disabledFrameworks.iterator().forEachRemaining(s -> sj.add(s));
+            String unrecognized = sj.toString();
+            throw new UserError(
+                    "The following argument(s) to the "
+                            + CalledMethodsChecker.DISABLE_BUILDER_FRAMEWORK_SUPPORTS
+                            + " command-line argument to the Called Methods Checker were unrecognized: "
+                            + unrecognized);
+        }
+
+        this.useValueChecker = checker.hasOption(CalledMethodsChecker.USE_VALUE_CHECKER);
+        this.collectionsSingletonList =
+                TreeUtils.getMethod(
+                        "java.util.Collections", "singletonList", 1, getProcessingEnv());
+        addAliasedAnnotation(OLD_CALLED_METHODS, CalledMethods.class, true);
+        addAliasedAnnotation(OLD_NOT_CALLED_METHODS, this.top);
+        this.postInit();
+    }
+
+    /**
+     * Enables support for the given builder-generation framework, unless it is listed in the
+     * disabled builder frameworks parsed from the -AdisableBuilderFrameworkSupport option's
+     * arguments.
+     *
+     * @param framework the builder framework to enable
+     * @param disabledFrameworks the set of disabled builder frameworks. This argument will be
+     *     side-effected to remove a builder framework if it was actually disabled.
+     */
+    private void enableFramework(String framework, Set<String> disabledFrameworks) {
```

Comment:
This method can be inlined (and specialized) where it is used, and I think the code will be simpler.

Reply:
I rewrote the whole method based on your other comments. I think it's clearer now as a separate method, but I can also inline it into the constructor if you still think that would be better.